packages/viz/src/lib/style/helpers/color-bins-style.ts

```diff
@@ -1,34 +1,72 @@
+import { RGBAColor } from '@deck.gl/aggregation-layers/utils/color-utils';
+
 import {
   getColors,
   getUpdateTriggers,
   hexToRgb,
   validateParameters
 } from './utils';
 
+import { Source } from '../../sources/Source';
+import { ClassificationMethod, Classifier } from '../Classifier';
+import { GeometryType, NumericFieldStats } from '../../types';
+import { Style } from '../Style';
+import { DefaultOptions, applyDefaults } from '../default-styles';
+
 export function colorBinsStyle(
   featureProperty: string,
-  {
-    bins = defaultOptions.bins,
-    colors = defaultOptions.colors,
-    nullColor = defaultOptions.nullColor,
-    othersColor = defaultOptions.othersColor
-  }: ColorBinsStyleOptions = defaultOptions
+  options?: ColorBinsStyleOptions
+) {
+  const opts = { ...defaultOptions, ...options };
+
+  validateBinParameters(featureProperty, opts.breaks, opts.palette);
+
+  const evalFN = (source: Source) => {
+    const geometryType = source.getGeometryType();
+
+    if (!opts.breaks.length) {
+      const stats = source.getFieldStats(featureProperty) as NumericFieldStats;
+      const classifier = new Classifier(stats);
+      const breaks = classifier.breaks(opts.bins, opts.method);
+      return calculateWithBreaks(featureProperty, breaks, geometryType, opts);
+    }
+
+    return calculateWithBreaks(
+      featureProperty,
+      opts.breaks,
+      geometryType,
+      opts
+    );
+  };
+
+  return new Style(evalFN, featureProperty);
+}
+
+function calculateWithBreaks(
+  featureProperty: string,
+  breaks: number[],
+  geometryType: GeometryType,
+  options: ColorBinsStyleOptions
 ) {
-  validateBinParameters(featureProperty, bins, colors);
+  const styles = applyDefaults(geometryType, options);
 
-  // Number.MIN_SAFE_INTEGER is here to make closed intervals,
-  // that way last range comparison will never be true
-  const ranges = [...bins, Number.MIN_SAFE_INTEGER];
+  // For 3 breaks, we create 4 ranges of colors. For example: [30,80,120]
+  // - From -inf to 29
+  // - From 30 to 79
+  // - From 80 to 119
+  // - From 120 to +inf
+  // Values lower than 0 will be in the first bucket and values higher than 120 will be in the last one.
+  const ranges = [...breaks, Number.MAX_SAFE_INTEGER];
 
   const {
     rgbaColors,
-    othersColor: rgbaOthersColor = hexToRgb(othersColor)
-  } = getColors(colors, bins.length - 1);
+    othersColor: rgbaOthersColor = hexToRgb(options.othersColor)
+  } = getColors(options.palette, ranges.length);
 
-  const rgbaNullColor = hexToRgb(nullColor);
+  const rgbaNullColor = hexToRgb(options.nullColor);
 
-  const getFillColor = (feature: Record<string, any>) => {
-    const featureValue: number = feature.properties[featureProperty];
+  const getFillColor = (feature: Record<string, any>): RGBAColor => {
```

Comment:
I guess you're changing this for compatibility, but I always found the way VL worked with bins not very intuitive

Reply:
I changed it because we had a bug. 

For the previous code if you have the following ranges [10,20,30], a value lower than 10 was assigned to a null color instead of the first interval

Reply:
Yes, I think that was on purpose, we thought on defining just closed intervals... in any case, I'm ok if the examples and tests are coherent